tardis/montecarlo/tests/test_cmontecarlo.py

```diff
@@ -1,99 +1,343 @@
 
-def test_compute_distance2boundary():
-	distance_to_boundary = 259376919351035.88
-	tests.test_compute_distance2boundary.restype = c_double
-	assert_almost_equal(tests.test_compute_distance2boundary(),
-		distance_to_boundary)
+@pytest.mark.parametrize(
+    ['packet_params', 'model_params', 'expected'],
+    [({'mu': 0.3, 'r': 7.5e14}, {'inverse_time_explosion': 1 / 5.2e7},
+      0.9998556693818854),
 
-def test_compute_distance2line():
-	distance_to_line = 7.792353908000001e+17
-	tests.test_compute_distance2line.restype = c_double
-	assert_almost_equal(tests.test_compute_distance2line(),
-		distance_to_line)
+     ({'mu': -.3, 'r': 8.1e14}, {'inverse_time_explosion': 1 / 2.6e7},
+      1.0003117541351274)]
+)
+def test_rpacket_doppler_factor(packet_params, model_params, expected, packet, model):
+    # Set the params from test cases here
+    packet.mu = packet_params['mu']
+    packet.r = packet_params['r']
+    model.inverse_time_explosion = model_params['inverse_time_explosion']
 
-def test_compute_distance2continuum():
-	distance_to_electron = 4.359272608766106e+28
-	tests.test_compute_distance2continuum.restype = c_double
-	assert_almost_equal(tests.test_compute_distance2continuum(),
-		distance_to_electron)
+    # Perform any other setups just before this, they can be additional calls
+    # to other methods or introduction of some temporary variables
+
+    # Set `restype` attribute if returned quantity is used
+    cmontecarlo_methods.rpacket_doppler_factor.restype = c_double
+    # Call the C method (make sure to pass quantities as `ctypes` data types)
+    obtained = cmontecarlo_methods.rpacket_doppler_factor(byref(packet), byref(model))
+
+    # Perform required assertions
+    assert_almost_equal(obtained, expected)
+
+
+@pytest.mark.parametrize(
+    ['packet_params', 'expected_params'],
+    [({'mu': 0.3, 'r': 7.5e14},
+      {'d_boundary': 259376919351035.88}),
+
+     ({'mu': -.3, 'r': 7.5e13},
+      {'d_boundary': -664987228972291.5}),
+
+     ({'mu': -.3, 'r': 7.5e14},
+      {'d_boundary': 709376919351035.9})]
+)
+def test_compute_distance2boundary(packet_params, expected_params, packet, model):
+    packet.mu = packet_params['mu']
+    packet.r = packet_params['r']
+
+    cmontecarlo_methods.compute_distance2boundary(byref(packet), byref(model))
+
+    assert_almost_equal(packet.d_boundary, expected_params['d_boundary'])
+
+
+@pytest.mark.parametrize(
+    ['packet_params', 'expected_params'],
+    [({'nu_line': 0.1, 'next_line_id': 0, 'last_line': 1},
+      {'tardis_error': 0, 'd_line': 1e+99}),
+
+     ({'nu_line': 0.2, 'next_line_id': 1, 'last_line': 0},
+      {'tardis_error': 0, 'd_line': 7.792353908000001e+17}),
+
+     ({'nu_line': 0.5, 'next_line_id': 1, 'last_line': 0},
+      {'tardis_error': 2, 'd_line': 0.0}),
+
+     ({'nu_line': 0.6, 'next_line_id': 0, 'last_line': 0},
+      {'tardis_error': 2, 'd_line': 0.0})]
+)
+def test_compute_distance2line(packet_params, expected_params, packet, model):
```

Comment:
I think we should test correctly returning errors in one test and the correct calculation of d_line in another. But this change can be done in a later iteration of these tests.